aptos-move/move-examples/mint_nft/sources/minting.move

```diff
@@ -0,0 +1,250 @@
+            vector<String>[string::utf8(b"address")],
+        );
+
+        let public_key = std::option::extract(&mut ed25519::new_validated_public_key_from_bytes(public_key_bytes));
+
+        move_to(origin, CollectionTokenMinter {
+            public_key,
+            signer_cap,
+            token_data_id,
+            expiration_timestamp,
+            minting_enabled: true,
+            token_minting_events: account::new_event_handle<TokenMintingEvent>(&resource),
+        });
+    }
+
+    /// Set if minting is enabled for this collection token minter
+    public entry fun set_minting_enabled(minter: &signer, minting_enabled: bool) acquires CollectionTokenMinter {
+        let minter_address = signer::address_of(minter);
+        assert!(minter_address == @mint_nft, error::permission_denied(ENOT_AUTHORIZED));
+        let collection_token_minter = borrow_global_mut<CollectionTokenMinter>(minter_address);
+        collection_token_minter.minting_enabled = minting_enabled;
+    }
+
+    /// Set the expiration timestamp of this collection token minter
+    public entry fun set_timestamp(minter: &signer, expiration_timestamp: u64) acquires CollectionTokenMinter {
+        let minter_address = signer::address_of(minter);
+        assert!(minter_address == @mint_nft, error::permission_denied(ENOT_AUTHORIZED));
+        let collection_token_minter = borrow_global_mut<CollectionTokenMinter>(minter_address);
+        collection_token_minter.expiration_timestamp = expiration_timestamp;
+    }
+
+    /// Mint an NFT to the receiver.
+    /// `mint_proof_signature` is the `MintProofChallenge` signed by the receiver's private key
+    /// `public_key_bytes` is the public key of the receiver
+    /// `account_scheme` is the account scheme of the receiver (should be 0/ed25519 or 1/multi_ed25519)
+    public entry fun mint_NFT(receiver: &signer, mint_proof_signature: vector<u8>, public_key_bytes: vector<u8>, account_scheme: u8) acquires CollectionTokenMinter {
+        let receiver_addr = signer::address_of(receiver);
+
+        // get the collection minter and check if the collection minting is disabled or expired
+        let collection_token_minter = borrow_global_mut<CollectionTokenMinter>(@mint_nft);
+
+        assert!(timestamp::now_seconds() < collection_token_minter.expiration_timestamp, error::permission_denied(ECOLLECTION_EXPIRED));
+        assert!(collection_token_minter.minting_enabled, error::permission_denied(EMINTING_DISABLED));
+
+        // verify that the receiver wants to claim this NFT
+        verify_proof_of_knowledge(receiver_addr, mint_proof_signature, collection_token_minter.token_data_id, public_key_bytes, account_scheme);
+
+        // mint token to the receiver
+        let resource_signer = account::create_signer_with_capability(&collection_token_minter.signer_cap);
+        token::mint_token_to(&resource_signer, receiver_addr, collection_token_minter.token_data_id, 1);
```

Comment:
This seems like the wrong paradigm as this requires explicit and not necessarily desirable opt-in.

I'd prefer for this contract we not increase scope and instead mint and then direct transfer (https://github.com/aptos-labs/aptos-core/blob/f7620e4c4608f6fa217d6986b732920950c6763b/aptos-move/framework/aptos-token/sources/token.move#L599)

We then should have a direct mint to function that if given both the minter and receiver signers can perform the appropriate one step mint / transfer.

Reply:
When the user claim the token, they already opt-in. We don't need offer and claim process in this airdrop scenario. 

I think mint_to require user opt-in direct transfer in this case. 
` 
opt_in_direct_transfer(receiver, true);
mint_to 
opt_in_direct_transfer(receiver, false);

`

Reply:
will test this later!